iceoryx_dds/include/iceoryx_dds/gateway/dds_to_iox.hpp

```diff
@@ -37,14 +38,32 @@ class DDS2IceoryxGateway : public gateway_t
     using ChannelFactory = std::function<channel_t(const capro::ServiceDescription)>;
 
   public:
+    /// @brief Default constructor
     DDS2IceoryxGateway() noexcept;
+
+    /// @brief Set DDS to iceoryx gateway object
+    /// @param[in] channelFactory the channel factory to propagate data across
     DDS2IceoryxGateway(ChannelFactory channelFactory) noexcept;
+
+    /// @brief Load the provided gateway configuration
+    /// @param[in] config the gateway configuration to load
     void loadConfiguration(const config::GatewayConfig& config) noexcept;
```

Comment:
What happens when it fails to load the configuration?

Reply:
That's basically also a very general question. Should the description mention possible exit points?
In this example it sets up a new channel. Looks like `loadConfiguration` might not be a suitable name for this method.

Reply:
There's no check for a valid config here. It's currently used in a successful expected. There's no error evaluation of `setupChannel()` here. Mentioned in comment.